### ISSUE TITLE
Add Right-to-Left editing

### ISSUE THREAD

**Issue by omidgolparvar** [Microsoft/vscode#4994]
Please add a feature to change writing direction.
For example, I want to edit markdown documents, and my language is Persian/Farsi, a Right-to-Left language. Now, when I edit documents, always it is left aligned and direction is left-to-right, and it is not comfortable to edit documents that have both Persian text and English text. 

Thanks.

**Comment by akosyakov** [comment]
It would be nice to have it for Monaco as well.

**Comment by Javanmardi** [comment]
Something like Notepad++ would be extremely sufficient because almost everyone needs to enter both of the at the same time.
In Notepad++ the combination of ALT+CTRL+R flips text to the right and ALT+CTRL+L reverse it.

**Comment by rafidka** [comment]
Any plan to support this at some point?

**Comment by TheEhsanSarshar** [comment]
I am working on this. hopefully I can find a way

**Comment by AhmedElTabarani** [comment]
@TheEhsanSarshar  how's it going

**Comment by yousufiqbal** [comment]
Its 2022. I'm amazed that it's not added..

**Comment by alatras** [comment]
Yes, we want this!

**Comment by Kikey-y** [comment]
It's been 6 years and this is still being ignored? Sad.

**Comment by DaudAhmad0303** [comment]
I got shocked that this feature is available even in **Notepad** but not in **VSCode**. Please add _**RTL**_ editing feature in VSCode.
![image](https://user-images.githubusercontent.com/73556387/192137180-41de3b61-42a3-4725-a0e7-191a1fda5944.png)

**Comment by mohammed-ezzedine** [comment]
such a shame this is not done yet

**Comment by mahdikiani** [comment]
And in 2023 it is not done yet!

**Comment by 411A** [comment]
The population of users who use RTL languages is more than 600 million! & we're begging to support this vital feature on every app every time! Obsidian please add RTL, VSCode please add RTL (really?! 7 years?!), etc. Let's hope that this issue will be resolved once and for all in the history of humanity.

**Comment by YoniChechik** [comment]
My hack for this problem: https://github.com/microsoft/vscode/issues/11770#issuecomment-1510465431

**Comment by w4po** [comment]
Almost 2024 and not done yet!!!, Just add it already, it's not even a complex feature!!

**Comment by kfkhalili** [comment]
Waiting on this. You can do it. You just have to believe in yourselves.

**Comment by ugcoder** [comment]
And in 2024 it is not done yet!

**Comment by mbaniasad** [comment]
this is a nice to have feature.

**Comment by mshannaq** [comment]
yes it is nice to have this feature

**Comment by massivefermion** [comment]
Very weird that it's being ignored like this!

**Comment by Mahsa-scs** [comment]
> icient because almost everyone needs to enter both of the languages at the same time. Web developers felt it so much.
> In Notepad++ the combination of ALT+CTRL+R flips text to the right and ALT+CTRL+L reverse it. I find it extremely sufficient instead of wise text orientating like Emacs or TeXstudio. As muc

Its 2024 and still not working!

**Comment by Mahsa-scs** [comment]
https://github.com/microsoft/vscode/issues/11770

**Comment by Mahsa-scs** [comment]
petition
https://chng.it/S99jnGn7KM

**Comment by mshannaq** [comment]
> VS Code is currently unusable with any RTL content!

yes it is not stable with the RTL text
i think the vc developers should take care of this

**Comment by MuziCodes** [comment]
There is [an extension](https://marketplace.visualstudio.com/items?itemName=SarbandAkray.rtl) that provides support for RTL if you name your file as example.rtl.txt (add .rtl before the extension)

**Comment by mshannaq** [comment]
> There is [an extension](https://marketplace.visualstudio.com/items?itemName=SarbandAkray.rtl) that provides support for RTL if you name your file as example.rtl.txt (add .rtl before the extension)

I will give it a try

**Comment by mohandex** [comment]
And in 2025 it is not done yet!

**Comment by ShlomiRex** [comment]
Please add support.

**Comment by xyasharx** [comment]
+1

**Comment by AhmedElTabarani** [comment]
Should we delegate this to Copilot if that would be faster, so you can just move on to the review iteration on it?
